safe/load.go

```diff
@@ -15,19 +17,45 @@ func Load(password []byte, backend backends.Client, encryptionClient crypto.Clie
 		password: password,
 	}
 
-	ciphertext, err := s.backend.Load()
-
-	if _, ok := err.(*errors.SafeNotFound); ok {
-		s.Accounts = make(map[string]Account)
-		return &s, nil
-	}
+	data, err := s.backend.Load()
 	if err != nil {
+		if _, ok := err.(*errors.SafeNotFound); ok {
+			s.Accounts = make(map[string]Account)
+			return &s, nil
+		}
 		return nil, err
 	}
 
-	plaintext, err := s.crypto.Decrypt(ciphertext, password)
+	safeFile := safeFile{}
+	if err := json.Unmarshal(data, &safeFile); err == nil {
+		// Unmarshalling did succeed -> Safe uses new format
+		// Do nothing
+	} else {
+		safeFile.Ciphertext = data
+		// If we don't have a config, use OpenPGP for backwards-compatibility
+		defaultOpenPGPConfig := crypto.DefaultConfigWithType("aes_openpgp")
+		safeFile.Config = &defaultOpenPGPConfig
+	}
+
+	upgradeNeeded := false
+	plaintext, err := s.crypto.Decrypt(safeFile.Ciphertext, password)
```

Comment:
I actually _was_ able to reproduce this now when using `type = "aes_openpgp"`and `cipher = "aes192"`.
It works fine for `cipher = "aes256"` and `cipher = "aes128"`.